### PR TITLE
authentication

### PR DESCRIPTION
**Pull request by AyaQunoo**

### CODE REVIEW

File: server/database/config/connection.js [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
   dialectOptions: {
     charset: 'utf8',
   },
+  ssl: {
+    require: true,
+    rejectUnauthorized: NODE_ENV !== 'development',
+  },
 });
-// sequelize.sync();
+sequelize.sync();
```
Comment (devstm) [inline server/database/config/connection.js]: recommit this line or try to but in in if() body based on NODE_ENV variable

File: server/routes/index.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 const router = require('express').Router();
 const languageRouter = require('./languageRouter');
+const authRouter = require('./auth');
 
 router.use('/lang', languageRouter);
```
Comment (devstm) [inline server/routes/index.js]: i know its not your change but lang is not descriptive name for this router
sorry :(

File: server/__test__/test.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (devstm) [inline server/__test__/test.js:1]: Try to test the end-points in two scenarios for each end-point

File: server/controllers/User/login.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    });
+
+    if (!user) {
+      throw new CustomeError('Email doesnt exists', 401);
+    }
+    const { id, username, password } = user;
+    const comparePassword = await compare(pass, password);
+
+    if (!comparePassword) {
+      return res.json('password incorrect');
```
Comment (devstm) [inline server/controllers/User/login.js]: throw unAuth Error :1st_place_medal:

File: server/controllers/User/login.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+const { compare } = require('bcrypt');
+const { loginValidation } = require('../../utils/validation/user');
+const { SignToken } = require('../../utils/jwt/jwt');
+const { CustomeError } = require('../../utils/helper/customeError');
```
Comment (devstm) [inline server/controllers/User/login.js]: use the index.js in utils insted of ../../utils/validation/user

File: server/controllers/User/login.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+const User = require('../../models/user');
```
Comment (devstm) [inline server/controllers/User/login.js]: create index file in models folder

File: server/controllers/User/oauth.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    const { email, name } = ticket.getPayload();
+
+    const existingUser = User.findOne({
+      where: {
+        email,
+      },
+    });
+    if (!existingUser) {
+      // eslint-disable-next-line no-unused-vars
```
Comment (devstm) [inline server/controllers/User/oauth.js]: if you use error first approach you dont need 
      // eslint-disable-next-line no-unused-vars
      anymore

File: server/controllers/User/oauth.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      const user = await User.create({
+        username: name,
+        email,
+      });
+    }
+    const token = await SignToken({
+      id: existingUser.id,
```
Comment (devstm) [inline server/controllers/User/oauth.js]: if there is no existingUser its shoud take the new user was created so try to handle that

File: server/controllers/User/oauth.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      name,
+      email,
+    });
+    return res.cookie('token', token)
+      .json({
+        msg: 'authenticated sucssfully',
```
Comment (devstm) [inline server/controllers/User/oauth.js:30]: return the user data

File: server/controllers/User/oauth.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        status: 201,
+      });
+  } catch (error) {
+    return res.status(401).json({ message: 'Authentication failed' });
```
Comment (devstm) [inline server/controllers/User/oauth.js]: return status

**Review by devstm — CHANGES_REQUESTED** [review]

**Review by MohammedMahani — APPROVED** [review]

**Review by abdullah-alsharif — APPROVED** [review]

**Review by Basel-01 — APPROVED** [review]

**Pull request merged**